PID-Analyzer.py

```diff
@@ -513,71 +512,79 @@ def beheader(self, loglist):
             heads.append(headsdict)
         return heads
 
-    def logfinder(self, fpath):
-        path, file = os.path.split(fpath)
-        name = file[:-3]
-        flist = os.listdir(path)
-        csvlist =   []
-        csv_sizes = []
-        eventlist =  []
-
-        for i in range(1, 99, 1):
-            csvname = name+'_temp'+str(i)+'.01.csv'
-            eventname = name+'_temp'+str(i)+'.01.event'
-            if csvname not in flist: break
-            csvlist.append(csvname)
-            eventlist.append(eventname)
-        for csv in csvlist:
-            csv_sizes.append(os.path.getsize(path+'/'+csv))
-        return csvlist, csv_sizes, eventlist
-
     def decode(self, fpath):
-        log = open(fpath, 'rb')
-        log2 = open(fpath, 'r')
-        firstline = log2.readlines()[0]
-        content = log.read()
-
-        split = content.split(str(firstline))
-        temps = []
+        """Splits out one BBL per recorded session and converts each to CSV."""
+        with open(fpath, 'r') as text_log_view:
+            # The first line of the overall BBL file re-appears at the beginning
+            # of each recorded session.
+            firstline = text_log_view.readlines()[0]
+        with open(fpath, 'rb') as binary_log_view:
+            content = binary_log_view.read()
+
+        split = content.split(firstline)
+        bbl_sessions = []
         for i in range(len(split)):
-            newfile=open(fpath[:-4]+'_temp'+str(i)+fpath[-4:], 'wb')
-            newfile.write(firstline+split[i])
-            temps.append(fpath[:-4]+'_temp'+str(i)+fpath[-4:])
-            newfile.close()
+            path_root, path_ext = os.path.splitext(os.path.basename(fpath))
+            temp_path = os.path.join(
+                self.tmp_dir, '%s_temp%d%s' % (path_root, i, path_ext))
+            with open(temp_path, 'wb') as newfile:
+                newfile.write(firstline+split[i])
+            bbl_sessions.append(temp_path)
 
         loglist = []
-        for t in temps:
-            size = os.path.getsize(self.path+'/'+t)
-            if size>500000:
+        for bbl_session in bbl_sessions:
+            size_bytes = os.path.getsize(os.path.join(self.tmp_dir, bbl_session))
+            if size_bytes > LOG_MIN_BYTES:
                 try:
-                    msg = os.system('blackbox_decode.exe' + ' '+t)
-                    loglist.append(t)
+                    msg = os.system(self.blackbox_decode_bin_path + ' '+bbl_session)
```

Comment:
I got a problem with spaces in the bbl_session path. 
I solved it with quotation marks: 
`msg = os.system(self.blackbox_decode_bin_path + ' ' + '"' + bbl_session + '"' )`

Reply:
The `subprocess` module is generally preferred to `os.system`. I switched it, and that should take care of quoting shell arguments as necessary. (I tested the new code with `crazy file name with " \ ' | .BBL` so hopefully if it deals that right it'll do the right thing on Windows too!)